lib/installation-proxy/index.js

```diff
@@ -50,13 +52,54 @@ class InstallationProxyService {
     this.plistService.sendPlist(request);
     const messages = await this.waitMessageCompletion();
     return messages.reduce((acc, message) => {
+      if (!message.CurrentList) {
+        return acc;
+      }
       message.CurrentList.forEach(app => {
         acc[app.CFBundleIdentifier] = app;
       });
       return acc;
     }, {});
   }
 
+  /**
+ * @typedef {Object} LookupApplicationOptions
+ *
+ * @property {string} applicationType of the which group you want to list. These can be User, System or leave it empty for both
+ * @property {Array} returnAttributes the fields which should be filtered and returned to the client. Leave this parameter empty if you don't want to filter
+ * @property {string|Array} bundleIds Bundle Ids of the apps that should be searched
+ */
+
+  /**
+   * Lists applications according to the opts and returns them as a map
+   * @param {LookupApplicationOptions} opts the lookup options that wants to be passed
+   * @returns A map of the applications which the key is the bundleId
+   */
+  async lookupApplications (opts = {}) {
+    const request = {
+      Command: 'Lookup',
+      ClientOptions: {}
+    };
+    if (opts.bundleIds) {
+      request.ClientOptions.BundleIDs = _.isString(opts.bundleIds) ? [opts.bundleIds] : opts.bundleIds;
+    }
+    if (opts.applicationType) {
+      request.ClientOptions.ApplicationType = opts.applicationType;
+    }
+    if (opts.returnAttributes) {
+      request.ClientOptions.ReturnAttributes = opts.returnAttributes;
+    }
+
+    this.plistService.sendPlist(request);
+    const messages = await this.waitMessageCompletion();
+    for (const message of messages) {
+      if (message.LookupResult) {
+        return message.LookupResult;
```

Comment:
Will only one message ever have a `LookupResult`?

Reply:
Yes, that is correct. I could do `messages[0]. LookupResult` but I like the for loop check better

Reply:
Me too.